src/js/front-end/menus/comments-menu.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Internal dependencies.
+ */
+import { domReady } from '../utils'; // Global utils.
+import { MENU_OPEN_CLASS_NAME } from './consts'; // Menu constants.
+import { createOverlay, removeOverlay } from './utils'; // Menu utils.
+
+// A class name to append to the body element when the comments menu is open.
+const openClassName = MENU_OPEN_CLASS_NAME + 'comments-menu';
+
+domReady( function () {
+	/**
+	 * Comments Menu toggle and overlay JavaScript.
+	 */
+	const body = document.body,
+		headerContain = document.querySelector( '.comments-menu' ),
+		commentsToggle = document.querySelectorAll( '.comments-menu-toggle' ),
+		commentsContents = document.querySelector( '.comments-contents' );
```

Comment:
```suggestion
		commentsContents = document.querySelector( '.comments-contents' ),
		commentsExist    = document.querySelector( '.wp-block-comments' );
		
		if ( ! commentsExist ) {
		    headerContain.parentNode.removeChild( headerContain );
		    return;
	    }
```

This could probably use more testing/cases, but it looks like it would work in a pinch to remove the comment toggle button -- it checks whether the `wp-block-comments` CSS class exists, and if not, removes the button. 

Another approach could be to add a CSS class to the body if comments are open and using that to show/hide things? My only concern with relying on something existing is if it doesn't work in all cases (though the above -- plus this setup in general -- does work with Disqus).